### PR TITLE
CP-12088: Fix crash when user tries to navigate within a browser tab

### PR DESCRIPTION
**Pull request by atn4z7**
## Description

**Ticket: [CP-12088]**

This PR fixes state mutation issues in the tab history reducer. Previously, we were mutating selector outputs (e.g., tab.historyIds = …), which caused runtime crashes due to Redux Toolkit’s immutability feature.

**Change:**
- Replaced direct mutations with updateOne calls to ensure immutable state updates.
- Added ESLint rule `no-param-reassign` to catch unsafe mutations.
- Added TypeScript typings with Readonly to enforce immutability at compile time for all the redux data adapters.

## Screenshots/Videos

**Before:**
https://github.com/user-attachments/assets/887c7116-a201-4723-a692-ae7cfa1b20f3

**After:**
https://github.com/user-attachments/assets/8a38b48e-87c2-4a65-8975-5e22464dc1ad

## Testing

**Dev Testing (if applicable)**

- Provide steps to test the happy path of your feature
- Provide steps to test edge cases and error states
- Trigger a build on bitrise and reference it here
- Move the ticket into the "Testing" column on Jira

**QA Testing (if applicable)**

- Provide instructions for QA to test this feature thoroughly
- State expected behavior / acceptance criteria

## Checklist

Please check all that apply (if applicable)

- [ ] I have performed a self-review of my code
- [ ] I have verified the code works
- [ ] I have included screenshots / videos of android and ios
- [ ] I have added testing steps
- [ ] I have added/updated necessary unit tests
- [ ] I have updated the documentation

### CODE REVIEW

File: packages/core-mobile/app/new/features/swap/hooks/useNavigateToSwap.ts [hunk 1]
```diff
@@ -8,8 +8,14 @@ export const useNavigateToSwap = (): {
...
-      fromTokenId = AVAX_TOKEN_ID
-      toTokenId = USDC_AVALANCHE_C_TOKEN_ID
+      navigate({
+        // @ts-ignore TODO: make routes typesafe
+        pathname: '/swap',
+        params: {
+          initialTokenIdFrom: AVAX_TOKEN_ID,
+          initialTokenIdTo: USDC_AVALANCHE_C_TOKEN_ID
+        }
+      })
```
Comment (B0Y3R-AVA) [inline packages/core-mobile/app/new/features/swap/hooks/useNavigateToSwap.ts:18]: do we need a return after this navigate to prevent the next one from firing right after the if check?

**Review by B0Y3R-AVA — APPROVED** [review]

**Review by B0Y3R-AVA — APPROVED** [review]

**Review by onghwan — APPROVED** [review]
dev-tested
